spatial/r3/mat_test.go

```diff
@@ -230,3 +230,60 @@ func BenchmarkQuat(b *testing.B) {
 		}
 	}
 }
+
+func TestMatHessian(t *testing.T) {
+	const (
+		tol = 1e-5
+		h   = 8e-4
+	)
+	step := Vec{X: h, Y: h, Z: h}
+	rnd := rand.New(rand.NewSource(1))
+	for _, Case := range []struct {
+		fn   func(p Vec) float64
+		want func(p Vec) *Mat
+	}{
```

Comment:
```suggestion
	for _, test := range matHessianTests {
```
and move the table up into package scope.